components/brave_wallet/browser/hd_keyring.h

```diff
@@ -29,50 +27,27 @@ class HDKeyring {
   HDKeyring(const HDKeyring&) = delete;
   HDKeyring& operator=(const HDKeyring&) = delete;
 
-  virtual void ConstructRootHDKey(const std::vector<uint8_t>& seed,
-                                  const std::string& hd_path);
+  static std::string GetRootPath(mojom::KeyringId keyring_id);
 
-  std::vector<AddedAcountInfo> AddAccounts(size_t number);
-  // This will return vector of address of all accounts
-  std::vector<std::string> GetAccountsForTesting() const;
-  // Only support removing accounts from the back to prevents gaps
-  void RemoveAccount();
+  virtual std::optional<AddedAccountInfo> AddNewHDAccount() = 0;
+  virtual void RemoveLastHDAccount() = 0;
 
+  // TODO(apaymyshev): imported accounts should be handled by separate keyring
```

Comment:
This is a highlight that imported keys should not be a part of HDKeyring. All the methods(with the exception of EncodePrivateKeyForExport) in this file are either HD or imported account specific. When they get separated - HDKeyring and implementation might have even simplier inteface requiring only derivation index instead of address and no `account_` state but just a cache, so no `AddNewHDAccount` `RemoveLastHDAccount` also